test/e2e/bandwidth_test.go

```diff
@@ -41,10 +42,14 @@ func TestBenchmarkBandwidthIntraNode(t *testing.T) {
 	if err := data.createPodOnNode("perftest-b", masterNodeName(), perftoolImage, nil, nil, nil, []v1.ContainerPort{{Protocol: v1.ProtocolTCP, ContainerPort: iperfPort}}, false); err != nil {
 		t.Fatalf("Error when creating the perftest server Pod: %v", err)
 	}
-	podBIP, err := data.podWaitForIP(defaultTimeout, "perftest-b", testNamespace)
+	podBIPs, err := data.podWaitForIP(defaultTimeout, "perftest-b", testNamespace)
 	if err != nil {
 		t.Fatalf("Error when getting the perftest server Pod's IP: %v", err)
 	}
+	if podBIPs.ipv4 == nil {
+		t.Fatal("Error when retrieving valid IPv4 address")
+	}
```

Comment:
I'm not sure both checks are required since you know that the cluster supports IPv4 at this stage and `podWaitForIP` didn't return an error.

At the very least, you can unify the checks so it's less verbose:
```go
if err != nil || podBIPs.ipv4 == nil
```

Reply:
I would move the existence of IPv4 address in function `podWaitForIP`, and remove the check of "podBIPs.ipv4 == nil" here.